analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -93,13 +93,14 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
 
     private static bool IsParameterDereferenced(IOperationWrapperSonar operation) =>
         operation.Parent != null
-        && operation.Parent.IsAnyKind(
+        && (operation.Parent.IsAnyKind(
             OperationKindEx.Invocation,
             OperationKindEx.FieldReference,
             OperationKindEx.PropertyReference,
             OperationKindEx.EventReference,
             OperationKindEx.Await,
-            OperationKindEx.ArrayElementReference);
+            OperationKindEx.ArrayElementReference)
+         || (operation.Parent.Kind == OperationKindEx.Conversion && IsParameterDereferenced(operation.Parent.ToSonar())));
```

Comment:
The `||` operator should be indented one level further w.r.t. the `&&` operator above (to which is not aligned either, by one char).
I think however, that it would be better to simplify the structure of this logical condition, since you have different logical operators at different levels of nesting.